Framework/Geometry/src/Objects/MeshObject.cpp

```diff
@@ -0,0 +1,725 @@
+#include "MantidGeometry/Objects/MeshObject.h"
+
+#include "MantidGeometry/Objects/Track.h"
+#include "MantidGeometry/Rendering/CacheGeometryHandler.h"
+#include "MantidGeometry/Rendering/GeometryHandler.h"
+#include "MantidGeometry/Rendering/vtkGeometryCacheReader.h"
+#include "MantidGeometry/Rendering/vtkGeometryCacheWriter.h"
+#include "MantidKernel/Exception.h"
+#include "MantidKernel/make_unique.h"
+#include "MantidKernel/Material.h"
+#include "MantidKernel/PseudoRandomNumberGenerator.h"
+
+#include <boost/make_shared.hpp>
+
+namespace Mantid {
+namespace Geometry {
+
+using Kernel::Material;
+using Kernel::V3D;
+using Kernel::Quat;
+
+/**
+*  Default constuctor
+*/
+MeshObject::MeshObject() : MeshObject("") {}
+
+MeshObject::MeshObject(const std::string &string)
+    : m_boundingBox(), m_object_number(0), m_handler(), m_string(string),
+      m_id(), m_material(), m_triangles(), m_vertices() {}
+
+/**
+* Copy constructor
+* @param A :: The object to initialise this copy from
+*/
+MeshObject::MeshObject(const MeshObject &A) : MeshObject() { *this = A; }
+
+/**
+* Assignment operator
+* @param A :: Object to copy
+* @return *this
+*/
+MeshObject &MeshObject::operator=(const MeshObject &A) {
+  if (this != &A) {
+    m_boundingBox = A.m_boundingBox;
+    m_object_number = A.m_object_number;
+    m_string = A.m_string;
+    m_handler = A.m_handler->clone();
+    m_id = A.m_id;
+    m_material = Kernel::make_unique<Material>(A.material());
+    m_triangles = A.m_triangles;
+    m_vertices = A.m_vertices;
+  }
+  return *this;
+}
+
+MeshObject::~MeshObject() = default;
+
+/*
+ * Initialise with vectors.
+ * @param faces :: Triangles specified by vertex index in anticlockwise order
+ * as seen from outside.
+ * @param vertices :: Vertex positions
+ */
+void MeshObject::initialize(const std::vector<int> &faces,
```

Comment:
Do we really need this public method? Why are these arguments not provided to the constructor instead? 

I notice that's there is quite a lot of code protecting for corrupted/incomplete state.

Reply:
Is there a good reason to use a signed type as the type argument to std::vector<T>? I think this should be unsigned since you are referring to point indices.